src/App.tsx

```diff
@@ -5,7 +5,18 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const handleAddMovie = (movie: Movie) => {
+    setMovies(prevState => {
+      const movieExist = prevState.find(state => state.imdbId === movie.imdbId);
```

Comment:
Prefer using meaningful variable name instead of 'state' in the find method. It would be more appropriate to use 'existingMovie' or 'currentMovie'.